code/modules/projectiles/projectile_parent.dm

```diff
@@ -738,6 +738,8 @@ ABSTRACT_TYPE(/datum/projectile)
 
 	var/static/effect_amount = 0
 
+	var/energy_particle_types = list(D_ENERGY, D_BURNING, D_RADIOACTIVE, D_TOXIC)
```

Comment:
This should not be here, the previous logic was much better. Defining this here will waste memory by initializing a list on every projectile.